src/lib/api/contrib/invenio/InvenioSearchApi.js

```diff
@@ -39,10 +40,14 @@ export class InvenioSearchApi {
   }
 
   initSerializers(config) {
+    this.requestSerializerMap = {
+      "InvenioRecordsResourcesRequestSerializer" : InvenioRecordsResourcesRequestSerializer,
+      "InvenioRequestSerializer" : InvenioRequestSerializer
+    }
     const requestSerializerCls = _get(
       config,
-      'invenio.requestSerializer',
-      InvenioRequestSerializer
+      this.requestSerializerMap[config['invenio.requestSerializer']],
+      InvenioRecordsResourcesRequestSerializer
```

Comment:
Here, I am not sure how to decide which `Serializer` to put to check the type with.

Reply:
I am not sure what you mean. Here you should probably not change everything, it was already configurable.
What are you trying to achieve here?

It should have as default `InvenioRequestSerializer` otherwise it should be passed via config.

Reply:
1. The change at `line 49` is because of this error mentioned below, so I have to create a map to an actual function instead of a string.

When I am configuring the `data-invenio-search-config` in `search.html` for the new request adapter. I tried with these two configs:
- First: I get `Uncaught TypeError: requestSerializerCls is not a constructor` since here I am sending it as a string.
```
"searchApi": {
    "invenio": {
      "requestSerializer": "InvenioRecordsResourcesRequestSerializer"
    }
  }
```
- Second: I get `Uncaught SyntaxError: JSON.parse: unexpected character` since it is without quotes.
```
"searchApi": {
    "invenio": {
      "requestSerializer": InvenioRecordsResourcesRequestSerializer
    }
  }
```

2. For the `line 50`, if I am keeping `InvenioRequestSerializer`, this one is taken into consideration and the facets formats is like before. Please let me know if I am missing something here.

Reply:
This is the right version indeed:

```
"searchApi": {
    "invenio": {
      "requestSerializer": InvenioRecordsResourcesRequestSerializer
    }
  }

```
but you cannot import it because you are in the `search.html`.
@zzacharo any idea how we can solve this? I would avoid the map, not flexible enough.

@ParthS007 I saw that in the `search.html` there is a spelling issue with Languages if you want to fix it:
```
      "title": "Laguanges"
```

Reply:
okay, I will include this in my PR.

Reply:
@ntarocco @ParthS007 There could be a solution to use `eval('ClassAsAString')` when we instantiate the `requestSerializer` in RSK (See https://stackoverflow.com/questions/49042459/how-to-instantiate-a-class-from-a-string-in-javascript). For that to work, you will need to import your custom class in the app-rdm bundle... That needs some experimentation though to see if the order of the bundle inclusion in the page is correct so RSK find the custom serializer class...

Reply:
I wouldn't go for eval if possible, maybe dynamic import?

@ParthS007 maybe the best thing to do is:
1. in `invenio` contrib, create a new function that will (pseudocode):
```
function getSerializer (serializer) {
    if serializer is string:
        return mappings[serializer]
    else
         return serializer // it is a already a class
}
2. in the InvenioSearchApi.js, call `getSerializer(serializer)

So if you pass a string, will go through a mapping as you have already done, but if you pass already a class/obj, then it will simply instantiate it.
Does this make sense?

Reply:
Yes, I think this sounds good and better for error handling as well. 

Pinging @jrcastro2, since he took over the issue now.

Reply:
Yes, I will go with this solution, thanks!